### PR TITLE
fix: go.mod: Use quic-go v0.25.0 for go1.18

### PR DESCRIPTION
**Pull request by jefft0**
Changes for go1.18.  Similar to the [Berty pull request](https://github.com/berty/berty/pull/4135), In go/go.mod, add a replace for quic-go v0.25. In packages/go.mod, change to quic-go v0.25. Run `go mod tidy` to update the go.sum files. 

Signed-off-by: jefft0 <jeff@thefirst.org>

### CODE REVIEW

**Comment by jefft0** [comment]
The code compiles for go1.18 with the existing dnsconfig_deprecated.go (with `go:build !go1.17`). We do not get the following error which we get for go1.17 without the hack:

    go.info.net.resolvConf: relocation target go.info.github.com/ipfs-shipyard/gomobile-ipfs/go/bind/core.resolverConfig not defined

File: go/bind/core/node.go [hunk 1]
```diff
@@ -38,6 +38,18 @@ type Node struct {
 }
 
 func NewNode(r *Repo, driver ProximityDriver) (*Node, error) {
+	var dialer net.Dialer
+	net.DefaultResolver = &net.Resolver{
+		PreferGo: false,
+		Dial: func(context context.Context, _, _ string) (net.Conn, error) {
+			conn, err := dialer.DialContext(context, "udp", "1.1.1.1:53")
```
Comment (aeddi) [inline go/bind/core/node.go:45]: Can you use the same DNS address as here? https://github.com/ipfs-shipyard/gomobile-ipfs/blob/9574047413bb2bf4446e8ae105fa11ddd35e934a/go/bind/core/dnsconfig.go#L79

**Review by aeddi — CHANGES_REQUESTED** [review]

File: go/bind/core/node.go (hunk 1 above)
Comment (jefft0) [reply go/bind/core/node.go]: I pushed the change. We only use one address. Is that OK?

Comment (aeddi-sudo) [reply go/bind/core/node.go]: Ok if we can't set a secondary address 👍

Comment (jefft0) [reply go/bind/core/node.go]: I looked at the golang source code to see if DialContext allows multiple addresses (maybe separated by comma), but it does not.
https://github.com/golang/go/blob/master/src/net/dial.go#L213

Merging with only the primary address.

**Pull request merged**